client/post-editor/edit-post-status/index.jsx

```diff
@@ -107,9 +107,7 @@ const EditPostStatus = React.createClass( {
 
 		if ( this.props.post ) {
 			isSticky = this.props.post.sticky;
-			isPending = this.props.savedPost.status === 'pending' || (
-				this.props.savedPost.status === 'draft' &&
-				this.props.post.status === 'pending' );
+			isPending = postUtils.isPending( this.props.post );
```

Comment:
So to truly do the same logic as you had before, this check should be against `savedPost` here, and the other clause could be changed to ` || ( this.props.savedPost.status === 'draft' && postUtils.isPending( this.props.post) `.  I suppose we could also add an `isDraft` utility as I'm guessing there are quite a few checks like this elsewhere.  But could be done in a different PR.

Looking just a few lines down, there are also some refactoring options to use postUtils more in this component.